tests/interfaces/test_batch_torch_qnode.py

```diff
@@ -0,0 +1,1099 @@
+
+        res1 = circuit(a)
+        grad_fn = qml.grad(circuit)
+        grad1 = grad_fn(a)
+
+        # switch to Torch interface
+        circuit.interface = "torch"
+
+        a = torch.tensor(0.1, dtype=torch.float64, requires_grad=True)
+
+        res2 = circuit(a)
+        res2.backward()
+        grad2 = a.grad
+        assert np.allclose(res1, res2.detach().numpy(), atol=tol, rtol=0)
+        assert np.allclose(grad1, grad2, atol=tol, rtol=0)
+
+    def test_drawing(self, dev_name, diff_method, mode):
+        """Test circuit drawing when using the torch interface"""
+
+        x = torch.tensor(0.1, requires_grad=True)
+        y = torch.tensor([0.2, 0.3], requires_grad=True)
+        z = torch.tensor(0.4, requires_grad=True)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qnode(dev, interface="torch")
+        def circuit(p1, p2=y, **kwargs):
+            qml.RX(p1, wires=0)
+            qml.RY(p2[0] * p2[1], wires=1)
+            qml.RX(kwargs["p3"], wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.probs(wires=0), qml.var(qml.PauliZ(1))
+
+        circuit(p1=x, p3=z)
+
+        result = qml.draw(circuit)(p1=x, p3=z)
+        expected = """\
+ 0: ──RX(0.1)───RX(0.4)──╭C──┤ Probs  
+ 1: ──RY(0.06)───────────╰X──┤ Var[Z] 
+"""
+
+        assert result == expected
+
+    def test_jacobian(self, dev_name, diff_method, mode, mocker, tol):
+        """Test jacobian calculation"""
+        if diff_method == "parameter-shift":
+            spy = mocker.spy(qml.gradients.param_shift, "transform_fn")
+        elif diff_method == "finite-diff":
+            spy = mocker.spy(qml.gradients.finite_diff, "transform_fn")
+
+        a_val = 0.1
+        b_val = 0.2
+
+        a = torch.tensor(a_val, dtype=torch.float64, requires_grad=True)
+        b = torch.tensor(b_val, dtype=torch.float64, requires_grad=True)
+
+        dev = qml.device(dev_name, wires=2)
+
+        @qnode(dev, diff_method=diff_method, mode=mode, interface="torch")
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))]
+
+        res = circuit(a, b)
+
+        assert circuit.qtape.trainable_params == {0, 1}
+
+        assert isinstance(res, torch.Tensor)
+        assert res.shape == (2,)
+
+        expected = [np.cos(a_val), -np.cos(a_val) * np.sin(b_val)]
+        assert np.allclose(res.detach().numpy(), expected, atol=tol, rtol=0)
+
+        loss = torch.sum(res)
+
+        loss.backward()
+        expected = [
+            -np.sin(a_val) + np.sin(a_val) * np.sin(b_val),
+            -np.cos(a_val) * np.cos(b_val),
+        ]
+        assert np.allclose(a.grad, expected[0], atol=tol, rtol=0)
+        assert np.allclose(b.grad, expected[1], atol=tol, rtol=0)
+
+        if diff_method in ("parameter-shift", "finite-diff"):
+            spy.assert_called()
+
+    @pytest.mark.xfail

```

Comment:
Why is this one `xfail`? (Looks like the tensorflow case is similar)

Reply:
oh! I remember now... it is because `qml.execute()` does not permit a dtype parameter (yet). I will update the changelog.